src/libtriton/arch/x86/x86Semantics.cpp

```diff
@@ -16563,44 +16563,29 @@ namespace triton {
         auto& dst  = inst.operands[0];
         auto& src  = inst.operands[1];
         bool  dstT = this->taintEngine->isTainted(dst);
-        bool  srcT = this->taintEngine->isTainted(src);
 
         /* Create symbolic operands */
         auto op1 = this->symbolicEngine->getOperandAst(inst, dst);
         auto op2 = this->symbolicEngine->getOperandAst(inst, src);
 
         /* Create the semantics */
-        auto node1 = op2;
-        auto node2 = op1;
-
-        /* Create symbolic expression */
-        auto expr1 = this->symbolicEngine->createSymbolicExpression(inst, node1, dst, "XCHG operation");
-        auto expr2 = this->symbolicEngine->createSymbolicExpression(inst, node2, src, "XCHG operation");
-
-        /* Spread taint */
-        expr1->isTainted = this->taintEngine->setTaint(dst, srcT);
-        expr2->isTainted = this->taintEngine->setTaint(src, dstT);
-
-        /* Create symbolic operands */
-        op1 = this->symbolicEngine->getOperandAst(inst, dst);
-        op2 = this->symbolicEngine->getOperandAst(inst, src);
-
-        /* Create the semantics */
-        auto node3 = this->astCtxt->bvadd(op1, op2);
+        auto node = this->astCtxt->bvadd(op1, op2);
```

Comment:
The `xadd` instruction started to fail on tests after implementation of destination memory operand check. The semantics was correct but somehow the symbolic expressions for the dst operand (in case of memory operand) was wrong. So I decided to rewrite semantics without dividing its logic on two steps (xchg and add) and calling `createSymbolicExpression()` only when final semantics is built - that fixes the tests.